src/main/java/com/google/javascript/clutz/AliasMapBuilder.java

```diff
@@ -0,0 +1,67 @@
+package com.google.javascript.clutz;
+
+import com.google.javascript.rhino.Node;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * AliasMapBuilder builds a map from the imported name of a symbol (based on a goog.require) to the
+ * exported name of a symbol (based on the variable name and goog.module id), so symbols that are
+ * imported and reexported can be properly aliased.
+ */
+public class AliasMapBuilder extends ImportBasedMapBuilder {
+  @Override
+  protected Map<String, String> build(
+      String localModuleId, Node moduleBody, Set<String> knownGoogProvides) {
+    Map<String, String> aliasMap = new HashMap<>();
+    if (localModuleId == null) {
+      //TODO(lucassloan): handle goog.module.get()
+      return aliasMap;
+    }
+
+    for (Node statement : moduleBody.children()) {
+      if (isImportAssignment(statement)) {
+        // `const C = goog.require()` or
+        // `const C = goog.module.get()`
+        String importedModuleId = statement.getFirstFirstChild().getChildAtIndex(1).getString();
+        String variableName = statement.getFirstChild().getString();
+
+        String exportedSymbolName = buildWholeModuleExportSymbolName(importedModuleId);
+        //TODO(lucassloan): handle knownGoogProvides
+        // exports = Foo

```

Comment:
I am confused here because you are not actually looking for 'exports = Foo'. I understand that's the ultimate intention of the code, but it needs some more explanation.

Currently the map just contains module$exports$C = module$exports$x.y.z. I guess closure will resolve from exports.C to module$exports$C , but that needs explanation.

Reply:
Put a comment on the extracted method - thoughts?